### ISSUE TITLE
Deperecated method TLSv1_2_server_method

### ISSUE THREAD

**Issue by denzen84** [xverizex/telegram_bot#1]
Please pay attention for deprecated method. Could you update the code to avoid this warning?

```
[ 25%] Building C object libcreqhttp/src/CMakeFiles/creqhttp.dir/creqhttp.c.o
/root/telegram_api_for_bot/libcreqhttp/src/creqhttp.c: In function ‘creqhttp_init_connection’:
/root/telegram_api_for_bot/libcreqhttp/src/creqhttp.c:464:17: warning: ‘TLSv1_2_server_method’ is deprecated: Since OpenSSL 1.1.0 [-Wdeprecated-declarations]
  464 |                 method = TLSv1_2_server_method ();
      |                 ^~~~~~
In file included from /root/telegram_api_for_bot/libcreqhttp/src/../include/creqhttp.h:25,
                 from /root/telegram_api_for_bot/libcreqhttp/src/creqhttp.c:26:
/usr/include/openssl/ssl.h:2032:50: note: declared here
 2032 | OSSL_DEPRECATEDIN_1_1_0 __owur const SSL_METHOD *TLSv1_2_server_method(void);
      |                                                  ^~~~~~~~~~~~~~~~~~~~~
```

**Comment by xverizex** [comment]
Oh, thanks. When I saw last time which the protocol is uses, the tls 1.2 did. I will see what protocol is uses and refactor the code if it changed.

**Comment by xverizex** [comment]
Telegram is working on the old machines and need to support the old protocol which on new machines deprecated, but is uses anyway.

**Comment by xverizex** [comment]
I have not found which protocol is uses now, but when I tested it, then experimentally established that the tls 1.2 uses.

**Comment by denzen84** [comment]
I used for a long time [telebot](https://github.com/smartnode/telebot) and for now it works too. But it uses to many deprecated methods from 'curl' and old BotAPI. Also, compiling bots with newest gcc produces a lot of warnings and errors. Author have no time to support it.

Your C-based telegram bot library seems to be most situable for migration from 'telebot'. So, I wrote simple echobot based on your lib and it compiles with gcc 12 fine. Thanks for great work.
